desc/compute/_neoclassical.py

```diff
@@ -126,8 +138,10 @@ def _compute(
         simp=simp,
     )
     out = batch_map(fun, fun_data, surf_batch_size)
-
-    return grid.expand(out)
+    if expand_out:
+        assert out.ndim == 1, "Are you sure you want to expand to full grid?"
```

Comment:
why this change?

Reply:
We only expand quantities if that expansion were to enable them to broadcast with others on the grid.

Reply:
I need to check this. Because full grid expansion was needed in some places which is why I removed that assert statement.

Reply:
For example, we don't expand the ballooning stability quantities either. It is confusing for the user if the shape `(num nodes, num alpha, num pitch)`. The user always wants `(num rho, num alpha, num pitch)`.

Reply:
I think there is case where the output shape is (num_rho, num_alpha, num_wells, num_pitch) which looks like (num_rho, num_alpha, num_zeta, num_pitch) to grid compress and it compresses num_wells. Perhaps the right solution is to add an extra dimension but let me find an example where out.ndim == 1 is needed. 
I guess for this we can allow this change.